sdk/python/kfp/components/structures.py

```diff
@@ -305,12 +305,10 @@ class ComponentSpec(BaseModel):
         implementation: The implementation of the component. Either an executor
             (container, importer) or a DAG consists of other components.
     """
-    # TODO(ji-yaqi): Update to OrderedDict for inputs and outputs once we drop
-    # Python 3.6 support
```

Comment:
I forget the context on this TODO. Doesn't `Dict` guarantee order already?
Do check with @ji-yaqi though.

Reply:
Good catch, @chensun. I think you're right... explanation:

Dicts are officially ordered in >= Python 3.7. This is a nuanced change: Python `dict`s are ordered, but are not `OrderedDict`s. We accept `dict` (e.g., dictionaries without [the special `OrderedDict` methods](https://docs.python.org/3/library/collections.html#collections.OrderedDict)), so it seems like we should not needlessly narrow this type from the superclass `dict` to the subclass `typing.OrderedDict`.

In code:
```python
import typing
import collections

d = {"a": True, "b": False}
od = collections.OrderedDict({"a": True, "b": False})

assert isinstance(od, dict) # type-checking would pass with the looser type
assert not isinstance(d, typing.OrderedDict) # type-checking would fail with the narrower type
```

@ji-yaqi, do you agree? If so, I will revert.

Further reading: [Choosing Between OrderedDict and dict](https://realpython.com/python-ordereddict/#choosing-between-ordereddict-and-dict)

Reply:
Updated: I reverted from `Dict` to `OrderedDict` in [b72d22f](https://github.com/kubeflow/pipelines/pull/7440/commits/b72d22f72d1c974ba564049d8065b77183910201).

Please let me know if this is not desired.

Reply:
OrderedDict SGTM, thank you!